test/forms/BankDetailsFormProviderSpec.scala

```diff
@@ -150,20 +142,50 @@ class BankDetailsFormProviderSpec extends StringFieldBehaviours with BarsTestDat
 
     "not bind strings with characters" in {
       val result        = form.bind(Map(fieldName -> "abcdef")).apply(fieldName)
-      val expectedError = FormError(fieldName, invalidKey, Seq(Validation.accountNumberPattern.toString))
+      val expectedError = FormError(fieldName, invalidKey, Seq(Validation.accountNumberPattern))
       result.errors shouldEqual Seq(expectedError)
     }
 
     "not bind strings with less than 6 digit" in {
-      val result = form.bind(Map(fieldName -> "12 34   5")).apply(fieldName)
+      val result = form.bind(Map(fieldName -> "12345")).apply(fieldName)
 
-      result.errors shouldEqual Seq(FormError(fieldName, lengthKey, Seq(minLength)))
+      result.errors shouldEqual Seq(FormError(fieldName, invalidKey, Seq(Validation.accountNumberPattern)))
     }
 
     "not bind strings with more than 8 digit" in {
-      val result = form.bind(Map(fieldName -> "12 34 56 789")).apply(fieldName)
+      val result = form.bind(Map(fieldName -> "123456789")).apply(fieldName)
+
+      result.errors shouldEqual Seq(FormError(fieldName, invalidKey, Seq(Validation.accountNumberPattern)))
+    }
+  }
+
+  "Form" must {
+    "Accept valid form data" in {
+      val form = new BankDetailsFormProvider().apply().bind(buildFormData("123456", "12345678"))
 
-      result.errors shouldEqual Seq(FormError(fieldName, lengthKey, Seq(maxLength)))
+      form.hasErrors shouldEqual false
+      form.value shouldEqual Some(BankDetails("AccountName", "123456", "12345678"))
+    }
+
+    "Pad 6 digit account codes" in {
```

Comment:
nice set of tests.
The only one I would add would be a 7 digit account number test
maybe one with spaces or dashes and one that is just exactly 7 numbers

Reply:
On it